### PR TITLE
Added per-service named loggers

### PR DESCRIPTION
**Pull request by bigspider**

### CODE REVIEW

File: src/dataEntities/startStop.ts [hunk 1]
```diff
@@ -18,13 +19,20 @@ export abstract class StartStopService extends EventEmitter {
...
+
     /**
      * A service that requires starting and stopping.
      * Whoever constructs this service must start it before using it.
      * Whoever constructs this service must stop it after using it.
      */
     protected constructor(protected readonly name: string) {
         super();
+
+        this.logger = createNamedLogger(name);
```
Comment (yahgwai) [inline src/dataEntities/startStop.ts]: The names currently being used are of inconsistent formats. I think we should standardise them - documentation is enough. Something along the lines of "Names must be lower case, words separated by hypthons".

We can do a quick run time check that this is the case, and throw a configuration if necessary.

The problem with using the names as they are now is that they have spaces in them - which isnt nice for file names.

File: src/dataEntities/startStop.ts [hunk 2]
```diff
@@ -41,7 +49,7 @@ export abstract class StartStopService extends EventEmitter {
         // set started straight away to block the code below
         this.mStarting = true;
         await this.startInternal();
-        logger.info(`${this.name}: Started.`);
+        this.logger.info(`${this.name}: Started.`);
```
Comment (yahgwai) [inline src/dataEntities/startStop.ts]: You no longer need ${this.name} in any logs of this class, or of it's sub classes

File: src/watcher/garbageCollector.ts [hunk 3]
```diff
@@ -84,9 +83,9 @@ export class AppointmentStoreGarbageCollector extends StartStopService {
                 // an error here means that we were likely unable to collect all, or some of, the appointments
                 // consecutive errors could mean we have a systematic problem, or connection issues with the store
                 // in either case the problem is very serious - so we stop the GC
-                logger.error("GC: Unexpected error.");
-                logger.error(`GC: Consecutive errors: ${this.consecutiveErrors}.`);
-                logger.error(doh);
+                this.logger.error("GC: Unexpected error.");
```
Comment (yahgwai) [inline src/watcher/garbageCollector.ts]: Remove the "GC" string from the logs in this class

File: src/logger.ts [hunk 4]
```diff
@@ -15,22 +15,31 @@ const myFormat = format.printf(info => {
...
+
+export function createNamedLogger(name: string | null) {
+    const prefix = name !== null ? name + "-" : "";
+    const newLogger = createLogger({
+        level: "info",
+        format: combinedFormats,
+        transports: [
+            new transports.File({ dirname: logDir, filename: `${prefix}error.log`, level: "error" }),
+            new transports.File({ dirname: logDir, filename: `${prefix}info.log`, level: "info" }),
+            new transports.File({ dirname: logDir, filename: `${prefix}debug.log`, level: "debug" })
```
Comment (yahgwai) [inline src/logger.ts]: I'm not sure why, but this doesnt appear to be working. Logs are appearing for debug - but they only contain info messages. Not sure if this was introduced by this PR, might have been a bug already. If so raise a bug ticket for it separately.

File: package.json [hunk 5]
```diff
@@ -12,12 +12,12 @@
     "start-evm": "ganache-cli -d -l 15000000 --allowUnlimitedContractSize --defaultBalanceEther 1000000 --mnemonic 'myth like bonus scare over problem client lizard pioneer submit female collect'",
     "submodule": "git submodule init && git submodule update",
     "lint": "eslint --ext ts .",
-    "test": "NODE_ENV=test && mocha -r ts-node/register 'test/src/**/*.test.ts' --exit && npm run test-mocked",
+    "test": "export NODE_ENV=test; mocha -r ts-node/register 'test/src/**/*.test.ts' --exit && npm run test-mocked",
```
Comment (yahgwai) [inline package.json]: Did you find these exports were necessary in the end?

Comment (bigspider) [reply package.json]: I investigated further, it seems that "NODE_ENV=test <command>" works, but not "NODE_ENV=test && command"; with export both versions work, which is why I was getting more consistent behavior, as we had a mix.
I removed the export and the also the extra &&.

File: src/dataEntities/startStop.ts (hunk 1 above)
Comment (bigspider) [reply src/dataEntities/startStop.ts]: Done, using the regexp `/^[a-z0-9\-]+$/`, so letters, numbers and hyphens.

File: src/dataEntities/startStop.ts (hunk 2 above)
Comment (bigspider) [reply src/dataEntities/startStop.ts]: Removed.

File: src/watcher/garbageCollector.ts (hunk 3 above)
Comment (bigspider) [reply src/watcher/garbageCollector.ts]: Removed.

File: src/logger.ts (hunk 4 above)
Comment (bigspider) [reply src/logger.ts]: It seems to be working as intended to me: "-error" only contains errors, "-info" contains errors and infos, while "-debug" contains everything. (see for example `watcher-debug.log` that has all the three levels).

Comment (yahgwai) [reply src/logger.ts]: Yeah, that's true, I was conflating two things.

The other thing then is that we appear to logging with level: "info" - so why should the debug files be created at all?

File: src/dataEntities/startStop.ts (hunk 1 above)
Comment (yahgwai) [reply src/dataEntities/startStop.ts]: It's good feedback to include the offending name as part of the error message.

File: src/logger.ts (hunk 4 above)
Comment (bigspider) [reply src/logger.ts]: Not sure I get what you mean. "debug" is the most verbose level, and we currently don't have it in many places, but the watcher is using it to show detailed Event information, which is about 12 lines of output.
Or would you rather only have "error" and "info" to reduce the number of log files?

File: src/dataEntities/startStop.ts (hunk 1 above)
Comment (bigspider) [reply src/dataEntities/startStop.ts]: Added better feedback in b083c9975182a41a572013f1993a80357e4fb63f.

File: src/logger.ts (hunk 4 above)
Comment (yahgwai) [reply src/logger.ts]: Ah no, it's fine to have debug. I'm just surprised that the logs are showing up.

```
createLogger({
level: "info",
format: combinedFormats,
```

I've looked up the documentation for `logger: info` and apparently that's supposed to be the minimum that's being logged. So I was wondering why debug files were being created when we have the minimum set to 'info' above.

Comment (yahgwai) [reply src/logger.ts]: Anyways, I'm sure this is a separate issue from your PR

**Pull request merged**